climada/engine/impact.py

```diff
@@ -983,7 +983,7 @@ def write(group: h5py.Group, name: str, value: Any):
 
         def _str_type_helper(values: Collection):
```

Comment:
Is `Collection` the correct type? Does this not also allow for dictionaries and such? Just curious to learn the correct type hint.

Reply:
`Collection` is any sized, iterable container, see https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes. The type might actually be too restrictive, `Iterable` would probably suffice here.

Since `dict` implements `__contains__`, `__iter__`, and `__len__`, it counts as `Collection`. Note the comment in the `type_writers` dict below: We try to cover the special types like `str`, `dict` and `csr_matrix` (which all count as `Collection`) before falling back to the collection writer.